spring-data-jdbc/src/test/java/org/springframework/data/jdbc/core/JdbcAggregateTemplateUnitTests.java

```diff
@@ -99,16 +99,15 @@ public void callbackOnSave() {
 
 		SampleEntity first = new SampleEntity(null, "Alfred");
 		SampleEntity second = new SampleEntity(23L, "Alfred E.");
-		SampleEntity third = new SampleEntity(23L, "Neumann");
```

Comment:
Why do you think the behaviour tested by this class should change?

Reply:
That might take some time to explain. Let me try :)

Previously we utilized `JdbcAggregateChangeExecutionContext#populateIdsIfNecessary` to set Ids to aggregates and other stuff. This method used to return `null`, if the entity after execution of DML have not been changed (simple example of how entity can be changed - there could be an auto generated identified provided by DB). And then after that we assigned the version to the entity. The essense of the problem were the follisng - both populateIdsIfNecessary method, or AggregateChange#getEntity method return entity without any version, even if we assign it previously.

Now, `JdbcAggregateChangeExecutionContext` can store in `results` map `InsertExecutionResult` object, which contains an entity with version - that is necessary, because we want to avoid redundant version population, but here is the problem - method `JdbcAggregateChangeExecutionContext#populateIdsIfNecessary` (I have given in my PR the more human readable, at least from my perspective, name to the method - `getRootWithPopulatedFieldsFromExecutionResult`) should not return null anymore, even if entity did not changed, becuase only JdbcAggregateChangeExecutionContext knows the version that was assigned to an entity. If we would return null, then we will loose this version information - just to remind, `AggregateChange#getEntity` does not know anything about version assigned to given entity, so if we will allow to return null, we will fallback to initaill issue.

This test case relied on fact, that AggregateChangeExecutor#execute will return an entity that is stored inside `aggregateChange` - wich is an entity from mock , the `third` one, but now it is no longer the truth. Now, the inserted entity will always be taken from `InsertExecutionResult`, which is the `second` one. Thus this test needs to be a little bit changed 

I really did my best to explain it :) @schauder Jens, in case of any further questions - do reach me out, I will clarify everything :)

Reply:
The problem with this is the test is a very valid test for the working of callbacks. 
We need to continue to work with the third value returned by the mock in the original version of the test, because it is a value a user might return from a callback.

Reply:
Jens @schauder, quite, I got it. So lets segregate 2 things - the user would expect, that the return value of the `tamplate#save()` method will be the same as the entity, returned from `AfterSaveCallback`, right? I agree,and I have fixed this, thanks! But becuase of the reasons I have explained upwards, `AfterSaveCallback` will be populated with the `second`, not `third` entity. That thing will remain, right @schauder? :)

Reply:
>  `AfterSaveCallback` will be populated with the second, not third entity.

No, that is not acceptable. The callback have to behave like a pipeline. What comes out of one needs to go into the next. The version update should be a part of that pipeline, so it must get its initial value from one callback and its result must go into the next callback.

Reply:
Ok, I got it) So I have fixed this, Jens @schauder, can you please re-review the changes?)